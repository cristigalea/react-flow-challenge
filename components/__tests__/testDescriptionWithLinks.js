/* @noflow */
import React from 'react';
import {shallow, render} from 'enzyme';

import DescriptionWithLinks from '../descriptionWithLinks';

describe('<DescriptionWithLinks />', () => {
    const defaultDescription = 'I often visit https://www.bbc.co.uk, especially the https://www.bbc.co.uk/weather section';
    const defaultLink1 = {url: 'https://www.bbc.co.uk', text: 'the BBC website'};
    const defaultLink2 = {url: 'https://www.bbc.co.uk/weather', text: 'weather'};

    it('should display a description with no links', () => {
        const wrapper = shallow(<DescriptionWithLinks description="My test description" links={[]} />);
        expect(wrapper.text()).toEqual('My test description');
    });
    it('should display a description with links from the same domain', () => {
        const wrapper = render(<DescriptionWithLinks description={defaultDescription} links={[defaultLink1, defaultLink2]} />);
        expect(wrapper.text()).toEqual('I often visit the BBC website, especially the weather section');
    });
    it('should display a description with only the matched links changed', () => {
        const wrapper = render(<DescriptionWithLinks description={defaultDescription} links={[defaultLink2]} />);
        expect(wrapper.text()).toEqual('I often visit https://www.bbc.co.uk, especially the weather section');
    });
    it('should display a description with links apearing multiple times', () => {
        const description = `
            I often visit https://www.bbc.co.uk, especially the https://www.bbc.co.uk/weather section.
            I also visit https://www.bbc.co.uk for entertainment.
        `;
        const wrapper = render(<DescriptionWithLinks description={description} links={[defaultLink1, defaultLink2]} />);
        expect(wrapper.text()).toEqual(`
            I often visit the BBC website, especially the weather section.
            I also visit the BBC website for entertainment.
        `);
    });
    it('should match the snapshot', () => {
        const wrapper = render(<DescriptionWithLinks description={defaultDescription} links={[defaultLink1, defaultLink2]} />);
        expect(wrapper).toMatchSnapshot();
    });
});
