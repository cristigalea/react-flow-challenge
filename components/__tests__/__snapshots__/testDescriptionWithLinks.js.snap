// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionWithLinks /> should match the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "I often visit ",
            "next": Object {
              "attribs": Object {
                "href": "https://www.bbc.co.uk",
              },
              "children": Array [
                Object {
                  "data": "the BBC website",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "next": Object {
                "data": ", especially the ",
                "next": Object {
                  "attribs": Object {
                    "href": "https://www.bbc.co.uk/weather",
                  },
                  "children": Array [
                    Object {
                      "data": "weather",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "next": Object {
                    "data": " section",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "text",
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          Object {
            "attribs": Object {
              "href": "https://www.bbc.co.uk",
            },
            "children": Array [
              Object {
                "data": "the BBC website",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "next": Object {
              "data": ", especially the ",
              "next": Object {
                "attribs": Object {
                  "href": "https://www.bbc.co.uk/weather",
                },
                "children": Array [
                  Object {
                    "data": "weather",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "next": Object {
                  "data": " section",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": Object {
              "data": "I often visit ",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
          },
          Object {
            "data": ", especially the ",
            "next": Object {
              "attribs": Object {
                "href": "https://www.bbc.co.uk/weather",
              },
              "children": Array [
                Object {
                  "data": "weather",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "next": Object {
                "data": " section",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "href": "https://www.bbc.co.uk",
              },
              "children": Array [
                Object {
                  "data": "the BBC website",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "data": "I often visit ",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
            },
            "type": "text",
          },
          Object {
            "attribs": Object {
              "href": "https://www.bbc.co.uk/weather",
            },
            "children": Array [
              Object {
                "data": "weather",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "next": Object {
              "data": " section",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": Object {
              "data": ", especially the ",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "href": "https://www.bbc.co.uk",
                },
                "children": Array [
                  Object {
                    "data": "the BBC website",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "data": "I often visit ",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                "type": "tag",
              },
              "type": "text",
            },
            "type": "tag",
          },
          Object {
            "data": " section",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "href": "https://www.bbc.co.uk/weather",
              },
              "children": Array [
                Object {
                  "data": "weather",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "data": ", especially the ",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "href": "https://www.bbc.co.uk",
                  },
                  "children": Array [
                    Object {
                      "data": "the BBC website",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "data": "I often visit ",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                  "type": "tag",
                },
                "type": "text",
              },
              "type": "tag",
            },
            "type": "text",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
